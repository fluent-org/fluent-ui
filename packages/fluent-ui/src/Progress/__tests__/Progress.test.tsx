import * as React from 'react'
import * as renderer from 'react-test-renderer'
import 'jest-styled-components'

import { ThemeProvider, Progress } from '../..'

describe('Progress', (): void => {
  const theme = {}
  test('basic', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Progress />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('prop value', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Progress value={50} />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
