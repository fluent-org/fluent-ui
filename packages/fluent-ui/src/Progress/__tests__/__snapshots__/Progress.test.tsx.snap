// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should be controlled 1`] = `
".FProgress-root {
  height: 4px;
  overflow: hidden;
  position: relative;
  background-color: #dedede;
}
.FProgress-progress {
  top: 0;
  left: 0;
  width: 100%;
  bottom: 0;
  position: absolute;
  background: #0078d4;
  transition: transform 0.43s linear;
  transform-origin: left;
}
.FProgress-isIndeterminate {
  animation: FProgress-keyframes-indeterminateProgress 2.6s infinite;
  background: linear-gradient(to right,
    #dedede 0%,
    #0078d4 50%,
    #dedede 100%
  );
}
@keyframes FProgress-keyframes-indeterminateProgress {
  from {
    transform: translateX(-130%);
  }
  to {
    transform: translateX(100%);
  }
}"
`;

exports[`Progress should be support ref 1`] = `
".FProgress-root {
  height: 4px;
  overflow: hidden;
  position: relative;
  background-color: #dedede;
}
.FProgress-progress {
  top: 0;
  left: 0;
  width: 100%;
  bottom: 0;
  position: absolute;
  background: #0078d4;
  transition: transform 0.43s linear;
  transform-origin: left;
}
.FProgress-isIndeterminate {
  animation: FProgress-keyframes-indeterminateProgress 2.6s infinite;
  background: linear-gradient(to right,
    #dedede 0%,
    #0078d4 50%,
    #dedede 100%
  );
}
@keyframes FProgress-keyframes-indeterminateProgress {
  from {
    transform: translateX(-130%);
  }
  to {
    transform: translateX(100%);
  }
}"
`;

exports[`Progress should render a indeterminate progress 1`] = `
".FProgress-root {
  height: 4px;
  overflow: hidden;
  position: relative;
  background-color: #dedede;
}
.FProgress-progress {
  top: 0;
  left: 0;
  width: 100%;
  bottom: 0;
  position: absolute;
  background: #0078d4;
  transition: transform 0.43s linear;
  transform-origin: left;
}
.FProgress-isIndeterminate {
  animation: FProgress-keyframes-indeterminateProgress 2.6s infinite;
  background: linear-gradient(to right,
    #dedede 0%,
    #0078d4 50%,
    #dedede 100%
  );
}
@keyframes FProgress-keyframes-indeterminateProgress {
  from {
    transform: translateX(-130%);
  }
  to {
    transform: translateX(100%);
  }
}"
`;
